x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/expandable_section/expandable_section_results.tsx

```diff
@@ -127,6 +186,104 @@ export const ExpandableSectionResults: FC<ExpandableSectionResultsProps> = ({
   );
   const analysisType =
     jobConfig && jobConfig.analysis ? getAnalysisType(jobConfig.analysis) : undefined;
+
+  const generateDiscoverUrl = useCallback(
+    async (rowIndex: number) => {
+      const item = tableItems[rowIndex];
+
+      if (discoverLocator !== undefined) {
+        const url = await discoverLocator.getRedirectUrl({
+          indexPatternId: dataViewId,
+          timeRange: data.query.timefilter.timefilter.getTime(),
+          filters: data.query.filterManager.getFilters(),
+          query: {
+            language: SEARCH_QUERY_LANGUAGE.KUERY,
+            // Filter for all visible column values of supported types - except the results field values
```

Comment:
I did wonder if we should add all columns to the filter, but that could potentially be a lot of columns. This way the user can control the filter by showing / hiding columns from the grid, so I think this makes sense.